Sources/NIOCore/EventLoop.swift

```diff
@@ -551,6 +551,94 @@ public struct TimeAmount: Hashable, Sendable {
     }
 }
 
+/// Contains the logic for parsing time amounts from strings,
+/// and printing pretty strings to represent time amounts.
+extension TimeAmount: CustomStringConvertible {
+
+    /// Errors thrown when parsint a TimeAmount from a string
+    public enum ValidationError: Error, Equatable {
+        /// Can't parse the provided unit
+        case unsupportedUnit(String)
+
+        /// Can't parse the number into a Double
+        case invalidNumber(String)
+    }
+
+    /// Creates a TimeAmount from a string representation with an optional default unit.
+    ///
+    /// Supports formats like:
+    /// - "5s" (5 seconds)
+    /// - "100ms" (100 milliseconds)
+    /// - "42" (42 of default unit)
+    /// - "1 hr" (1 hour)
+    ///
+    /// This function only supports one pair of the number and units, i.e. "5s" or "100ms" but not "5s 100ms".
+    ///
+    /// Supported units:
+    /// - h, hr, hrs (hours)
+    /// - m, min (minutes)
+    /// - s, sec, secs (seconds)
+    /// - ms, millis (milliseconds)
+    /// - us, µs, micros (microseconds)
+    /// - ns, nanos (nanoseconds)
+    ///
+    /// - Parameters:
+    ///   - userProvidedString: The string to parse
+    ///   - defaultUnit: Unit to use if no unit is specified in the string
+    ///
+    /// - Throws: ValidationError if the string cannot be parsed
+    public init(_ userProvidedString: String, defaultUnit: String = "s") throws {
```

Comment:
Should this be `@inlinable` or at least `@usableFromInline`?

Reply:
`@usableFromInline` is not usable on `public` methods. I have no objection to making it `@inlinable`, but I don't think it's necessary either.